src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -199,9 +201,10 @@ impl<'a> JobQueue<'a> {
     // In general, we try to print "Compiling" for the first nontrivial task
     // run for a package, regardless of when that is. We then don't print
     // out any more information for a package after we've printed it once.
-    fn note_working_on(&mut self, config: &Config, pkg: &'a PackageId,
+    fn note_working_on(&mut self, config: &Config, key: &Key<'a>,
                        fresh: Freshness) -> CargoResult<()> {
-        if self.printed.contains(&pkg) {
+        if (self.compiled.contains(key.pkg) && key.profile.doc == false) || 
+            (self.documented.contains(key.pkg) && key.profile.doc == true) {
```

Comment:
It's generally more idiomatic to just have `if foo` rather than `if foo == true` or `if !foo` instead of `if foo == false` in these two boolean checks.